packages/core/js-client/src/clientPeer/types.ts

```diff
@@ -107,65 +112,66 @@ export interface IFluenceClient extends IFluenceInternalApi {
   getRelayPeerId(): string;
 }
 
-/**
- * Configuration used when initiating Fluence Client
- */
-export interface ClientConfig {
-  /**
-   * Specify the KeyPair to be used to identify the Fluence Peer.
-   * Will be generated randomly if not specified
-   */
-  keyPair?: KeyPairOptions;
-
-  /**
-   * Options to configure the connection to the Fluence network
-   */
-  connectionOptions?: {
+export const configSchema = z
+  .object({
     /**
-     * When the peer established the connection to the network it sends a ping-like message to check if it works correctly.
-     * The options allows to specify the timeout for that message in milliseconds.
-     * If not specified the default timeout will be used
+     * Specify the KeyPair to be used to identify the Fluence Peer.
+     * Will be generated randomly if not specified
      */
-    skipCheckConnection?: boolean;
-
+    keyPair: keyPairOptionsSchema,
     /**
-     * The dialing timeout in milliseconds
+     * Options to configure the connection to the Fluence network
      */
-    dialTimeoutMs?: number;
-
+    connectionOptions: z
+      .object({
+        /**
+         * When the peer established the connection to the network it sends a ping-like message to check if it works correctly.
+         * The options allows to specify the timeout for that message in milliseconds.
+         * If not specified the default timeout will be used
+         */
```

Comment:
interesting to know if these doc comments still appear in IDE in autocomplete as they previously did when zod was not used. If they are lost - I would think about somehow restoring them

Reply:
Yes, comments should be preserved. I checked